Firmware/ultralcd.cpp

```diff
@@ -5249,45 +5253,43 @@ static void lcd_main_menu()
     if (farm_mode)
         MENU_ITEM_FUNCTION_P(_T(MSG_FILAMENTCHANGE), lcd_colorprint_change);//8
 
-    if ( moves_planned() || printer_active() ) {
-        MENU_ITEM_SUBMENU_P(_T(MSG_TUNE), lcd_tune_menu);
-    } else if (!Stopped) {
-        MENU_ITEM_SUBMENU_P(_i("Preheat"), lcd_preheat_menu);////MSG_PREHEAT c=18
-        if (M79_timer_get_status()) {
-            if(GetPrinterState() == PrinterState::IsReady) {
-                MENU_ITEM_FUNCTION_P(_T(MSG_SET_NOT_READY), lcd_printer_ready_state_toggle);
-            } else {
-                MENU_ITEM_FUNCTION_P(_T(MSG_SET_READY), lcd_printer_ready_state_toggle);
+    if (!printer_recovering()) {
+        if ( moves_planned() || printer_active()) {
+            MENU_ITEM_SUBMENU_P(_T(MSG_TUNE), lcd_tune_menu);
+        } else if (!Stopped) {
+            MENU_ITEM_SUBMENU_P(_i("Preheat"), lcd_preheat_menu);////MSG_PREHEAT c=18
+            if (M79_timer_get_status()) {
+                if(GetPrinterState() == PrinterState::IsReady) {
+                    MENU_ITEM_FUNCTION_P(_T(MSG_SET_NOT_READY), lcd_printer_ready_state_toggle);
+                } else {
+                    MENU_ITEM_FUNCTION_P(_T(MSG_SET_READY), lcd_printer_ready_state_toggle);
+                }
             }
         }
-    }
-    if (mesh_bed_leveling_flag == false && homing_flag == false && !printingIsPaused() && !processing_tcode) {
-        if (usb_timer.running()) {
-            MENU_ITEM_FUNCTION_P(_T(MSG_PAUSE_PRINT), lcd_pause_usb_print);
-        } else if (IS_SD_PRINTING) {
-            MENU_ITEM_FUNCTION_P(_T(MSG_PAUSE_PRINT), lcd_pause_print);
+        if (mesh_bed_leveling_flag == false && homing_flag == false && !printingIsPaused() && !processing_tcode) {
+            if (usb_timer.running()) {
+                MENU_ITEM_FUNCTION_P(_T(MSG_PAUSE_PRINT), lcd_pause_usb_print);
+            } else if (IS_SD_PRINTING) {
+                MENU_ITEM_FUNCTION_P(_T(MSG_PAUSE_PRINT), lcd_pause_print);
+            }
         }
     }
-    if(printingIsPaused())
-    {
+    if (printingIsPaused()
         // only allow resuming if hardware errors (temperature or fan) are cleared
-        if(!get_temp_error()
+        && !get_temp_error()
 #ifdef FANCHECK
-            && fan_check_error != EFCE_REPORTED
+        && fan_check_error != EFCE_REPORTED
 #endif //FANCHECK
-           ) {
-            if (saved_printing) {
+        && (saved_printing_type != PowerPanic::PRINT_TYPE_NONE || saved_printing)
+        && custom_message_type != CustomMsg::Resuming) {
+        if (saved_printing_type == PowerPanic::PRINT_TYPE_SD) {
                 MENU_ITEM_SUBMENU_P(_T(MSG_RESUME_PRINT), lcd_resume_print);
-            } else {
+        } else if ((saved_printing_type == PowerPanic::PRINT_TYPE_HOST) && (M79_timer_get_status())) {
```

Comment:
hmm, isn't this a regression for users which **don't** use the M79 timer? If a user doesn't use M79, this change means they can no longer resume the print from the LCD. But still they can pause the print from LCD.

See `lcd_pause_usb_print` which doesn't have this restriction.

Reply:
It has been pointed out that the menu shouldn't be shown as long the host isn't connected.
Testing selected Resume too early while the RPi Zero was still booting up after a power panic. Nothing happened as the `//action:resume` never reached the host.
Using the `M79` ensures that the host is connected and ready to receive instructions like `//action:resume` and prevent users to select the menu too early.